src/main/java/com/minecolonies/coremod/colony/managers/StatisticAchievementManager.java

```diff
@@ -0,0 +1,273 @@
+        @NotNull final NBTTagCompound fishermanStatisticsCompound = new NBTTagCompound();
+        @NotNull final NBTTagCompound lumberjackStatisticsCompound = new NBTTagCompound();
+        compound.setTag(TAG_STATISTICS, statisticsCompound);
+        statisticsCompound.setTag(TAG_MINER_STATISTICS, minerStatisticsCompound);
+        minerStatisticsCompound.setInteger(TAG_MINER_ORES, minedOres);
+        minerStatisticsCompound.setInteger(TAG_MINER_DIAMONDS, minedDiamonds);
+        statisticsCompound.setTag(TAG_FARMER_STATISTICS, farmerStatisticsCompound);
+        farmerStatisticsCompound.setInteger(TAG_FARMER_CARROTS, harvestedCarrots);
+        farmerStatisticsCompound.setInteger(TAG_FARMER_POTATOES, harvestedPotatoes);
+        farmerStatisticsCompound.setInteger(TAG_FARMER_WHEAT, harvestedWheat);
+        statisticsCompound.setTag(TAG_GUARD_STATISTICS, guardStatisticsCompound);
+        guardStatisticsCompound.setInteger(TAG_GUARD_MOBS, killedMobs);
+        statisticsCompound.setTag(TAG_BUILDER_STATISTICS, builderStatisticsCompound);
+        builderStatisticsCompound.setInteger(TAG_BUILDER_HUTS, builtHuts);
+        statisticsCompound.setTag(TAG_FISHERMAN_STATISTICS, fishermanStatisticsCompound);
+        fishermanStatisticsCompound.setInteger(TAG_FISHERMAN_FISH, caughtFish);
+        statisticsCompound.setTag(TAG_LUMBERJACK_STATISTICS, lumberjackStatisticsCompound);
+        lumberjackStatisticsCompound.setInteger(TAG_LUMBERJACK_TREES, felledTrees);
+        lumberjackStatisticsCompound.setInteger(TAG_LUMBERJACK_SAPLINGS, plantedSaplings);
+    }
+
+    @Override
+    public void incrementStatistic(@NotNull final String stat, @NotNull final Colony colony)
+    {
+        final int statisticAmount = this.getStatisticAmount(stat);
+        incrementStatisticAmount(stat);
+        if (statisticAmount >= NUM_ACHIEVEMENT_FIRST)
+        {
+            TriggerColonyAchievements.triggerFirstAchievement(stat, colony);
+        }
+        if (statisticAmount >= NUM_ACHIEVEMENT_SECOND)
+        {
+            TriggerColonyAchievements.triggerSecondAchievement(stat, colony);
+        }
+        if (statisticAmount >= NUM_ACHIEVEMENT_THIRD)
+        {
+            TriggerColonyAchievements.triggerThirdAchievement(stat, colony);
+        }
+        if (statisticAmount >= NUM_ACHIEVEMENT_FOURTH)
+        {
+            TriggerColonyAchievements.triggerFourthAchievement(stat, colony);
+        }
+        if (statisticAmount >= NUM_ACHIEVEMENT_FIFTH)
+        {
+            TriggerColonyAchievements.triggerFifthAchievement(stat, colony);
+        }
+    }
+
+    @Override
+    public List<Advancement> getAchievements()
+    {
+        return new ArrayList<>(colonyAchievements);
+    }
+
+    /**
+     * Get the amount of statistic.
+     *
+     * @param statistic the statistic.
+     * @return amount of statistic.
+     */
+    private int getStatisticAmount(@NotNull final String statistic)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this method 11, down to the maximum allowed 6. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)